### PR TITLE
feat: Ensure rhosak gcp onlybilled in rhm

### PR DESCRIPTION
**Pull request by miguelsorianod**
## Description

Related to https://issues.redhat.com/browse/MGDSTRM-9370.

For Kafkas created in `gcp` Cloud Provider currently the only supported billing models are prepaid (standard) and consumption based pricing on redhat marketplace. This PR updates AMS quota logic to limit the billing models for Kafkas created in `gcp` to the supported billing models.

## Verification Steps
* Create a Kafka instance with `gcp` cloud provider with AWS marketplace billing model. An error should be returned.
* Create a Kafka instance with `gcp` cloud provider with RHM marketplace billing model. It should succeed (as long as there is enough quota)
* Create a Kafka instance with `gcp` cloud provider on prepaid billing model. it should succed (as long as there is enough quota)
* The behavior of creating Kafka instances with `aws` cloud provider does not change

Note: KFM should be run with the `quota-type` flag set to `ams`

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: internal/kafka/internal/services/quota/ams_quota_service.go [hunk 1]
```diff
@@ -359,6 +363,17 @@ func (q amsQuotaService) ReserveQuota(kafka *dbapi.KafkaRequest, instanceType ty
 	}
 	kafka.BillingModel = matchedBillingModel
 
+	// For Kafka requests to be provisioned on GCP currently the only supported
+	// AMS billing models are standard or Red Hat Marketplace ("marketplace").
+	// If the AMS billing model to be requested is none of those we return an error.
+	// TODO change the logic to send "marketplace-rhm" instead of the deprecated
+	// "marketplace" once the billing dependencies are updated to deal with it.
+	if kafka.CloudProvider == cloudproviders.GCP.String() &&
```
Comment (machi1990) [inline internal/kafka/internal/services/quota/ams_quota_service.go:371]: I remember we've a similar check for supported billing models: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/internal/handlers/validation.go#L32, that only checks for marketplace and standard.

I wonder if the check in this PR  could be  cloud provider agnostic as well?

Comment (miguelsorianod) [reply internal/kafka/internal/services/quota/ams_quota_service.go:371]: The check cannot be cloud provider agnostic.

This is checking an AMS biling model whereas the link you show is a Kafka billing model

Comment (machi1990) [reply internal/kafka/internal/services/quota/ams_quota_service.go:371]: Thanks for the clarification. 

So just so I understood correctly, the Kafka billing model indicates the payment option either prepaid (`standard`) or consumption based (`marketplace`). 

While the AMS billing model is also a payment option including pre-paid (`standard`) and a nuanced version of consumption based (`marketplace`): can be `marketplace-rhm`, `marketplace-aws`, `marketplace-azure` etc?

Comment (miguelsorianod) [reply internal/kafka/internal/services/quota/ams_quota_service.go:371]: That's correct.

Comment (machi1990) [reply internal/kafka/internal/services/quota/ams_quota_service.go:371]: Thanks.

**Review by machi1990 — APPROVED** [review]
The code change lgtm. Thanks for the explanation @miguelsorianod 
I'll prefer if this is reviewed by someone else as well so I've tagged @pb82

**Comment by pb82** [comment]
:eyes:

File: internal/kafka/internal/services/quota/ams_quota_service.go [hunk 2]
```diff
@@ -81,12 +81,15 @@ func (q amsQuotaService) newBaseQuotaReservedResourceBuilder(kafka *dbapi.KafkaR
...
 	rr.Count(1)
 	return *rr
 }
 
-var supportedAMSBillingModels map[string]struct{} = map[string]struct{}{
+// supportedAMSRelatedResourceBillingModels returns the supported AMS billing
+// models for AMS RelatedResources. An AMS RelatedResource is not to be confused
+// with an AMS ReservedResource. The set of Billing models shown/accepted is
+// different between them
+var supportedAMSRelatedResourceBillingModels map[string]struct{} = map[string]struct{}{
```
Comment (pb82) [inline internal/kafka/internal/services/quota/ams_quota_service.go:92]: double type declaration, can be shortened to `var supportedAMSRelatedResourceBillingModels = map[string]struct{}{`

Comment (miguelsorianod) [reply internal/kafka/internal/services/quota/ams_quota_service.go]: Changed

**Review by pb82 — APPROVED** [review]

**Pull request merged**